specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/stable/2021-10-01/Workspaces_NetworkSecurityPerimeter_API.json

```diff
@@ -0,0 +1,442 @@
+          "$ref": "#/definitions/NSPConfigPerimeter"
+        },
+        "resourceAssociation": {
+          "$ref": "#/definitions/NSPConfigAssociation"
+        },
+        "profile": {
+          "$ref": "#/definitions/NSPConfigProfile"
+        },
+        "provisioningIssues": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NSPProvisioningIssue"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "NSPConfigAccessRule": {
+      "description": "Information of Access Rule in Network Security Perimeter profile",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the resource",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/NSPConfigAccessRuleProperties"
+        }
+      }
+    },
+    "NSPConfigAccessRuleProperties": {
+      "description": "Properties of Access Rule",
+      "type": "object",
+      "properties": {
+        "direction": {
+          "description": "Direction of Access Rule",
+          "enum": [
+            "Inbound",
+            "Outbound"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "NspAccessRuleDirection",
+            "modelAsString": true
+          }
+        },
+        "addressPrefixes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "fullyQualifiedDomainNames": {
+          "description": "FQDN for outbound rules",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "description": "Fully qualified domain name",
+            "type": "string"
+          }
+        },
+        "subscriptions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "networkSecurityPerimeters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NSPConfigPerimeter"
+          }
+        }
+      }
+    },
+    "NSPConfigAssociation": {
+      "description": "Information about resource association",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "accessMode": {
+          "description": "Access Mode of the resource association",
+          "enum": [
+            "Enforced",
+            "Learning",
+            "Audit"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ResourceAssociationAccessMode",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "NSPConfigPerimeter": {
+      "description": "NetworkSecurityPerimeter related information",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "arm-id"
+        },
+        "perimeterGuid": {
+          "type": "string"
+        },
```

Comment:
My comment was showing up in another commit. 

name says guid, but type is string. can it be perimeterId ?
please make it consistent.

Reply:
we cannot change the names of the fields since these models need to be same across all services. For example, the storage service has this as well which has gone through arm review already:
https://github.com/Azure/azure-rest-api-specs/blob/cf230c94500f8ee044acc68cb7bab37eace5671d/specification/storage/resource-manager/Microsoft.Storage/stable/2023-04-01/networkSecurityPerimeter.json#L421

Reply:
I couldn't find examples of any open api specs that have guid types in the repo, can you point me to an example?